concurrency/mutexmap.go

```diff
@@ -106,6 +108,16 @@ func (a *mutexMap[T]) Delete(key T) {
 	a.lock.Unlock()
 }
 
+func (a *mutexMap[T]) DeleteUnlock(key T) {
+	a.lock.Lock()
+	mutex, ok := a.items[key]
+	delete(a.items, key)
```

Comment:
Shouldn't the item be deleted only if ok is true? Otherwise two calls to this function might see the first deleting, releasing the lock and then the second attemping to delete an already deleted item.

Reply:
I think this should be ok, but it may not hurt to do as you suggest anyways. Calling `delete` with a key that doesn't exist is a no-op.

Reply:
If it's a no-op then we should be good, thanks for confirming